src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java

```diff
@@ -92,13 +95,19 @@ public void onPost(String triggeredByUser) {
      * Called when a POST is made.
      */
     public void onPost(final GitHubTriggerEvent event) {
+        if (Objects.isNull(job)) {
+            return; // nothing to do
+        }
+
+        Job<?, ?> currentJob = notNull(job, "Job can't be null");
```

Comment:
Why is it needed?

Reply:
This was suggested by @jglick the dereferencing allows us to guarantee the job continues to be not-null for the rest of the execution block.